### DISCUSSION TITLE
Citation: how to make the automatic "dropdown" options after "@" work?

### DISCUSSION THREAD

**Discussion by hao-hong** [quarto-dev/quarto-cli#4202] [Q&A]
Hi all,

I am not sure if this is the best way to describe the question. A few days ago, I can create a citation in quarto by adding "@", and then there will be a dropdown menu showing available citation keys for me to choose. But this function suddenly does not work any more. I have to manually input the exact citation keys each time. Everything else works fine.

Can anyone help me here? Thank you.

Edit: I am using a `.bib` file. And the dropdown menu does not work in both VSCode and R-studio.

Edit 2: It turned out that I could do this on a different computer. So, it should have something to do with my laptop.

**Comment by dragonstyle** [comment]
That features uses the bibliography in the front matter of your document. Do you have a valid path to a bibliography file there?

**Reply by hao-hong** [reply]
I am not good at this. I just put the bib file in the same folder with the `.qmd` file. It worked before but stopped.

**Reply by dragonstyle** [reply]
No worries- happy you're using Quarto! You also need to have the bibliography listed in the document, like:

```
---
title: My Title
bibliography. mybib.bib
---

## Section

```

Is that there?

**Reply by hao-hong** [reply]
Yes. And I can export `.pdf` documents with citations and references without a problem. the only problem is that when I am editing the `.qmd` file, and if I type "@"+whatever, there is no drop down menu for me to select from a list of citation keys. I have to manually input the citation key myself.

Oddly, when I use my desktop, I can have the dropdown menu. But it stops working after like an hour. I did nothing but opened the `.bib` file.

**Reply by hao-hong** [reply]
For my own record, I do not know why. But if I put brackets "[" and "]" for the `.bib` file, the autocompletion for citations works well.

**Comment by mcanouil** [comment]
@hao-hong You are talking about auto-completion from the IDE which is unrelated to Quarto CLI itself.

Quarto extension for VSCode does not have this feature (yet).

RStudio does support this at least in the last version.
Quarto website has a page describing this feature in RStudio IDE, see <https://quarto.org/docs/visual-editor/technical.html>.
If you are using RStudio IDE, then you might want to update it and set it up as described in the link above.
